packages/webapp/src/_app/ui/header-nav.tsx

```diff
@@ -77,31 +77,27 @@ const HeaderItemLink = ({
 };
 
 const AccountMenu = () => {
+    const queryClient = useQueryClient();
     const [ualAccount, ualLogout, ualShowModal] = useUALAccount();
-    const [memberName, setMemberName] = useState("");
-
     const accountName = ualAccount ? ualAccount.accountName : undefined;
 
-    useEffect(() => {
-        const updateLoggedMemberName = async (account: string) => {
-            const member = await getEdenMember(account);
-            if (member) {
-                setMemberName(member.name);
-            }
-        };
-        if (ualAccount && ualAccount.accountName) {
-            updateLoggedMemberName(ualAccount.accountName);
-        } else {
-            setMemberName("");
-        }
-    }, [ualAccount, accountName]);
+    const { data: member } = useQuery(
+        ["current-member", accountName],
+        async () => await getEdenMember(accountName),
```

Comment:
React Query actually automatically passes variables in to the function it calls. I'll do a slight refactor to allow for that: https://react-query.tanstack.com/guides/query-functions#query-function-variables

This will end up looking like this:

```js
const { data: member } = useQuery(
        ["current-member", accountName],
        getEdenMember,
        { staleTime: Infinity, enabled: !!ualAccount }
);
```

But I'll have to modify `getEdenMember` slightly.